operator/pkg/ciliumendpointslice/controller_test.go

```diff
@@ -0,0 +1,135 @@
+			fakeClient = *c
+			ciliumEndpoint = cep
+			ciliumEndpointSlice = ces
+			return nil
+		}),
+	)
+	if err := hive.Start(context.Background()); err != nil {
+		t.Fatalf("failed to start: %s", err)
+	}
+	cesCreated, err := createCEPandVerifyCESCreated(fakeClient, ciliumEndpoint, ciliumEndpointSlice)
+	if err != nil {
+		t.Fatalf("Couldn't verify if CES is created: %s", err)
+	}
+	// Verify CES is created when CES features is disabled
+	assert.Equal(t, true, cesCreated)
+	if err := hive.Stop(context.Background()); err != nil {
+		t.Fatalf("failed to stop: %s", err)
+	}
+}
+
+func TestNotRegisterControllerWithCESDisabled(t *testing.T) {
+	defer goleak.VerifyNone(
+		t,
+		// To ignore goroutine started by the workqueue. It reports metrics
+		// on unfinished work with default tick period of 0.5s - it terminates
+		// no longer than 0.5s after the workqueue is stopped.
+		goleak.IgnoreTopFunction("k8s.io/client-go/util/workqueue.(*Type).updateUnfinishedWorkLoop"),
+	)
+	var fakeClient k8sClient.FakeClientset
+	var ciliumEndpoint resource.Resource[*cilium_v2.CiliumEndpoint]
+	var ciliumEndpointSlice resource.Resource[*cilium_v2a1.CiliumEndpointSlice]
+	h := hive.New(
+		k8sClient.FakeClientCell,
+		k8s.ResourcesCell,
+		cell.Provide(func() Config {
+			return defaultConfig
+		}),
+		cell.Provide(func() SharedConfig {
+			return SharedConfig{
+				EnableCiliumEndpointSlice: false,
+			}
+		}),
+		cell.Invoke(func(p params) error {
+			registerController(p)
+			return nil
+		}),
+		cell.Invoke(func(c *k8sClient.FakeClientset, cep resource.Resource[*cilium_v2.CiliumEndpoint], ces resource.Resource[*cilium_v2a1.CiliumEndpointSlice]) error {
+			fakeClient = *c
+			ciliumEndpoint = cep
+			ciliumEndpointSlice = ces
+			return nil
+		}),
+	)
+	if err := h.Start(context.Background()); err != nil {
+		t.Fatalf("failed to start: %s", err)
+	}
+	cesCreated, err := createCEPandVerifyCESCreated(fakeClient, ciliumEndpoint, ciliumEndpointSlice)
+	if err != nil {
+		t.Fatalf("Couldn't verify if CES is created: %s", err)
+	}
+	// Verify CES is NOT created when CES features is disabled
+	assert.Equal(t, false, cesCreated)
+	if err = h.Stop(context.Background()); err != nil {
+		t.Fatalf("failed to stop: %s", err)
+	}
+}
+
+func createCEPandVerifyCESCreated(fakeClient k8sClient.FakeClientset, ciliumEndpoint resource.Resource[*cilium_v2.CiliumEndpoint], ciliumEndpointSlice resource.Resource[*cilium_v2a1.CiliumEndpointSlice]) (bool, error) {
+	cep := createStoreEndpoint("cep1", "ns", 1)
+	fakeClient.CiliumV2().CiliumEndpoints("ns").Create(context.Background(), cep, meta_v1.CreateOptions{})
+	cepStore, _ := ciliumEndpoint.Store(context.Background())
+	if err := testutils.WaitUntil(func() bool {
+		return len(cepStore.List()) > 0
+	}, time.Second); err != nil {
+		return false, errors.Errorf("failed to get CEP: %s", err)
```

Comment:
I don't see a good argument for having `github.com/pkg/errors` as a new direct dependency. 
```suggestion
		return false, fmt.Errorf("failed to get CEP: %s", err)
```

Reply:
Fixed, thanks